compiler/rustc_errors/src/lib.rs

```diff
@@ -536,6 +538,13 @@ impl<'a> std::ops::Deref for DiagCtxtHandle<'a> {
     }
 }
 
+pub type HirDelayedLint = (
+    &'static Lint,
+    HirId,
+    Span,
+    Box<dyn DynSend + for<'a, 'b> FnOnce(&'b mut Diag<'a, ()>) + 'static>
```

Comment:
similarly this won't work, as you can't hash it for incremental

Reply:
should be fixed now